src/installer/corehost/cli/bundle/runner.cpp

```diff
@@ -81,3 +81,47 @@ bool runner_t::locate(const pal::string_t& relative_path, pal::string_t& full_pa
 
     return true;
 }
+
+bool STDMETHODCALLTYPE runner_t::bundle_probe(const wchar_t *path, int64_t *offset, int64_t *size)
+{
+    if (path == nullptr)
+    {
+        return false;
+    }
+
+    pal::string_t file_path;
+
+    if (!pal::unicode_palstring(path, &file_path))
+    {
+        trace::warning(_X("Failure probing contents of the application bundle."));
+        trace::warning(_X("Failed to convert path [%ls] to UTF8"), path);
+
+        return false;
+    }
+
+    const file_entry_t* entry = app()->probe(file_path);
+    if (entry != nullptr)
+    {
+        *offset = entry->offset();
+        *size = entry->size();
+
+        assert(*offset != 0);
+
+        return true;
+    }
+
+    return false;
+}
+
+pal::string_t runner_t::get_bundle_probe()
```

Comment:
Just my personal preference: I would move this to hostpolicy coreclr setup code. I don't think the runner is the right component to know about who we encode the probe value to pass it to the runtime - that's a concern which is solved by hostpolicy coreclr interaction code.